tensorboard/plugins/hparams/api.proto

```diff
@@ -0,0 +1,187 @@
+// a training program may generate one or more series of real
+// numbers--each containing the evaluation of some metric at different
+// training steps.
+//
+// In some experiments, the user trains a model with the same set of
+// hyperparameters multiple times to get the distribution of metric
+// evaluations when the computation (such as the training algorithm, or metric
+// evaluation) is non-deterministic. To make the UI aware of this, sessions
+// are partitioned into groups: each group consists of all training sessions
+// which share the same values for the hyperparameters. In experiments with no
+// repeated executions, each group consists of exactly one session.
+message Experiment {
+  // A unique id.
+  string name = 1;
+
+  string description = 2;
+
+  // An id for the owning user or group.
+  string user = 3;
+
+  google.protobuf.Timestamp time_created = 4;
+
+  // Information about each hyperparameter used in the experiment.
+  repeated HParamInfo hparam_infos = 5;
+
+  // Information about each metric used in the experiment.
+  repeated MetricInfo metric_infos = 6;
+}
+
+message HParamInfo {
+  // An id for the hyperparameter (unique within an experiment).
+  string name = 1;
+
+  // A string used to display the hyperparameter in the UI. If empty the UI
+  // will display the 'name' field.
+  string display_name = 2;
+
+  string description = 3;
+
+  // The list of distinct values this hyperparameter had in any session within
+  // the experiment. Currently, the UI considers hyperparameter values as opaque
+  // strings.
+  repeated string vals = 4;
+}
+
+message MetricInfo {
+  // A name for the MetricInfo. Within an experiment there should be at most
+  // one MetricInfo having the given name.
+  string name = 1;
+
+  // A string used to display the hyperparameter in the UI. If empty the UI
+  // will display the 'name' field.
+  string display_name = 2;
+
+  string description = 3;
+
+  // The dataset type (validation, training) on which the metric is computed.
+  DatasetType dataset_type = 4;
+}
+
+message SessionGroup {
+  string name = 1;
+  repeated HParamVal hparam_vals = 2;
+}
+
+message HParamVal {
+  string name = 1;
+  string value = 2;
+}
+
+message Session {
+  // An id for the session. Unique within an experiment.
+  string name = 1;
+
+  google.protobuf.Timestamp time_started = 2;
+
+  // May be empty if unavailable or the session has not finished yet.
+  google.protobuf.Timestamp time_completed = 3;
+
+  // May be STATUS_UNKNOWN if unavailable.
+  Status status = 4;
+
+  // A filesystem path that will allow the user to reconstruct the model for
+  // this session. E.g., in Tensorflow this should be the directory where
+  // checkpoints are saved.
+  string model_path = 5;
+
+  // An optional link to a website monitoring the session.
+  string monitor_url = 6;
+}
+
+enum DatasetType {
+  DATASET_UNKNOWN = 0;
+  DATASET_TRAINING = 1;
+  DATASET_VALIDATION = 2;
+}
+
+enum Status {
```

Comment:
Please add a comment: is this the Status of a Session? A Group? Both?

Reply:
Done.